include/boost/compute/context.hpp

```diff
@@ -175,27 +175,13 @@ class context
     /// devices, the first is returned.
     device get_device() const
     {
-        size_t count = 0;
-        clGetContextInfo(m_context,
-                         CL_CONTEXT_DEVICES,
-                         0,
-                         0,
-                         &count);
-        if(count == 0){
-            return device();
-        }
+        std::vector<device> devices = get_devices();
 
-        cl_device_id id;
-        clGetContextInfo(m_context,
-                         CL_CONTEXT_DEVICES,
-                         sizeof(cl_device_id),
-                         &id,
-                         0);
-        if(id == 0){
+        if(devices.empty()) {
             return device();
         }
 
-        return device(id);
+        return devices.front();
```

Comment:
Sorry for the delay, I've been quite busy. Could you rebase this on the latest `develop` branch and fix-up the commit messages? Should be fairly easy with an interactive rebase (`git rebase -i boostorg/develop`). Thanks!